berserk/clients/puzzles.py

```diff
@@ -78,3 +79,22 @@ def create_race(self) -> PuzzleRace:
         """
         path = "/api/racer"
         return cast(PuzzleRace, self._r.post(path))
+
+    def get_next(
+        self, angle: str | None = None, difficulty: Difficulty | None = None
+    ) -> PuzzleResponse:
+        """Get a random Lichess puzzle (optionally filtered by theme or difficulty).
+
+        If authenticated, will only return puzzles the user hasn't seen before.
+
+        :param angle: theme or opening to filter puzzles
+        :param difficulty: PuzzleDifficulty enum
+        :return: a new puzzle
+        """
+        path = "/api/puzzle/next"
+        params: dict[str, str] = {}
+        if angle:
+            params["angle"] = angle
+        if difficulty:
+            params["difficulty"] = difficulty.value
```

Comment:
this is already handled by `requests` which does not send payload key-value pairs when the value is None